cmd/geth/main.go

```diff
@@ -301,32 +326,58 @@ func startNode(ctx *cli.Context, stack *node.Node) {
 		utils.Fatalf("Unable to attach to node: %v", err)
 	}
 
+	// Perform addt'l unlock steps depending on whether this is a
+	// maker or voter.  Observers are handled by first unlock above.
 	var (
 		voteKey      *ecdsa.PrivateKey
 		blockVoteKey *ecdsa.PrivateKey
 	)
 
 	if addr := ctx.GlobalString(utils.VoteAccountFlag.Name); addr != "" {
+		usingVotePassword := ctx.GlobalIsSet(utils.VoteAccountPasswordFlag.Name)
+		if usingVault && usedVault {
+			utils.Fatalf("Specified more than one account to unlock.  This is unsupported with Vault passwords, use only one of --unlock, --voteaccount, or --blockmakeraccount.")
+		}
 		addr = strings.TrimSpace(addr)
 		var passwd []string
-		if ctx.GlobalIsSet(utils.VoteAccountPasswordFlag.Name) {
+		if usingVotePassword && usingVault {
+			utils.Fatalf("Specified a plaintext --votepassword while trying to use Vault Password, this is not supported.")
+		} else if usingVault {
+			passwd = passwords
+		} else if usingVotePassword {
 			passwd = append(passwd, ctx.GlobalString(utils.VoteAccountPasswordFlag.Name))
+		} else {
+			utils.Fatalf("Tried to unlock a voter account, but no password was provided via --votepassword or Vault password.")
 		}
 
 		unlockAccount(ctx, accman, addr, 0, passwd)
 		// unlockAccounts fatals in case the account could not be unlocked
+		if usingVault {
+			usedVault = true
+		}
 		voteKey, err = accman.Key(common.HexToAddress(addr))
 		if err != nil {
 			utils.Fatalf("Unable to unlock vote key: %v", err)
 		}
 	}
 
 	if addr := ctx.GlobalString(utils.VoteBlockMakerAccountFlag.Name); addr != "" {
+		usingMakerPassword := ctx.GlobalIsSet(utils.VoteBlockMakerAccountPasswordFlag.Name)
+		if usingVault && usedVault {
+			utils.Fatalf("Specified more than one account to unlock.  This is unsupported with Vault passwords, use only one of --unlock, --voteaccount, or --blockmakeraccount.")
+		}
 		addr = strings.TrimSpace(addr)
 		var passwd []string
-		if ctx.GlobalIsSet(utils.VoteBlockMakerAccountPasswordFlag.Name) {
+		if usingMakerPassword && usingVault {
+			utils.Fatalf("Specified more than one account to unlock.  This is unsupported with Vault passwords, use only one of --unlock, --voteaccount, or --blockmakeraccount.")
+		} else if usingVault {
+			passwd = passwords
+		} else if usingMakerPassword {
 			passwd = append(passwd, ctx.GlobalString(utils.VoteBlockMakerAccountPasswordFlag.Name))
+		} else {
+			utils.Fatalf("Tried to unlock a voter account, but no password was provided via --blockmakerpassword or Vault password.")
```

Comment:
Tried to unlock a *blockmaker* account, but no password was provided via --blockmakerpassword or Vault password.

Reply:
Most recent commit fixes this